### PR TITLE
Feature/update contribution

### PR DESCRIPTION
**Pull request by blondered**
## Description

Updated contribution guide

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Optimization

## How Has This Been Tested?
Before submitting a PR, please check yourself against the following list. It would save us quite a lot of time.
- Have you read the contribution guide?
- Have you updated the relevant docstrings? We're using Numpy format, please double-check yourself
- Does your change require any new tests?
- Have you updated the changelog file?

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -7,7 +7,7 @@ new submissions, or contributing code that can be incorporated into the
 project.
 
 Please, don't use the issue tracker for support questions. Instead use:
```
Comment (feldlime) [inline CONTRIBUTING.rst:9]: We have a `question` issue type, maybe we have to change this line?

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -27,13 +27,36 @@ Repository as a Pull Request.
...
+      install everything that is needed with poetry. See `poetry usage details <https://python-poetry.org/docs/basic-usage/#installing-with-poetrylock>`_
+   
+#. Implement your changes. Check the following after you are done:
+   
+   #. Code styling. Autoformat with ``make format``
+   #. Linters. Run checks with ``make lint``
+   #. Tests. Make sure you've covered new features with tests. Check
+      code correctness with ``make test``
+   #. Coverage. Check with ``make coverage``
+   #. Docstrings. Please ensure that all new public classes and methods
```
Comment (feldlime) [inline CONTRIBUTING.rst:49]: This point is covered with `make lint`.

We can remove it or put to the beginning of this list.

File: CONTRIBUTING.rst (hunk 1 above)
Comment (blondered) [reply CONTRIBUTING.rst]: fixed

File: CONTRIBUTING.rst (hunk 2 above)
Comment (blondered) [reply CONTRIBUTING.rst]: moved to the beginning

**Review by feldlime — APPROVED** [review]

**Pull request merged**